src/ui/public/filter_bar/filter_bar.js

```diff
@@ -28,13 +29,35 @@ define(function (require) {
           'invertFilter',
           'invertAll',
           'removeFilter',
-          'removeAll'
+          'removeAll',
+          'updateFilter'
         ].forEach(function (method) {
           $scope[method] = queryFilter[method];
         });
 
         $scope.state = getAppState();
 
+        $scope.aceLoaded = function (editor) {
+          editor.$blockScrolling = Infinity;
+          var session = editor.getSession();
+          session.setTabSize(2);
+          session.setUseSoftTabs(true);
+
+          session.on('changeAnnotation', function checkIfValid() {
```

Comment:
Generally, with external event emitters you need to call `$scope.$digest()` or something similar. Is that not the case here?